lib/cdo/share_filtering.rb

```diff
@@ -51,20 +48,12 @@ module FailureType
   # @param [String] program the student's program text
   # @param [String] locale a two-character ISO 639-1 language code
   def self.find_share_failure(program, locale, exceptions: false)
-    return nil unless should_filter_program(program)
```

Comment:
Currently, when `find_share_failure` is called, it only checks programs that are `studio` game type, i.e., Play Lab apps AND include indicators of user-generated text.

Reply:
`find_share_failure` is used in:
- `activities_controller`'s [`milestone`](https://github.com/code-dot-org/code-dot-org/blob/213cd29df1e39781b6863fd9ec510536c25a8a1a/dashboard/app/controllers/activities_controller.rb#L59),
- `evaluate_rubric_job`'s [`perform`](https://github.com/code-dot-org/code-dot-org/blob/213cd29df1e39781b6863fd9ec510536c25a8a1a/dashboard/app/jobs/evaluate_rubric_job.rb#L243),
-

Reply:
For the evaluation job, I think `ShareFiltering.find_failure` would suffice instead of `find_share_failure`. The only difference between those two is that `find_share_filter` removes the tag names inside xml/html and, as discussed here, looks for app indicators. Should be fine to not do those things in the AI eval job.

Reply:
We switched to `find_failure` here: https://github.com/code-dot-org/code-dot-org/pull/65630